src/snek5000/output/phys_fields.py

```diff
@@ -1,73 +1,100 @@
-"""Paraview interface for Nek5000 files.
-
-.. todo:: Load field files and make various plots of statistics
+"""Interface for Nek5000 field files. :class:`PhysFields` usually
+instantiated as ``sim.output.phys_fields`` provides a common interface for
+plotting and reading arrays from the solution field files. Loading of data
+files are managed by the classes in the :mod:`snek5000.output.readers`.
 
 """
-from ..log import logger
+from functools import cached_property
 
-try:
-    import paraview.simple as pvs
-    from paraview.simple import VisItNek5000Reader
-except ImportError as e:
-    logger.warning(e)
+from fluidsim_core.params import iter_complete_params
 
+#  from .readers import try_paraview_ as pv
+from .readers import pymech_ as pm
 
-class PhysFields:
-    """Class for loading, plotting simulation files.
 
-    :param sim: A simulation instance derived from
-                :class:`snek5000.solvers.base.SimulNek`
-    :param path: :class:`pathlib.Path` Path where simulation state files are
-                 found.
+class PhysFields:
+    """Class for loading, plotting simulation files."""
+
+    @staticmethod
+    def _complete_info_solver(info_solver, classes=None):
+        """Static method to complete the ParamContainer info_solver.
+
+        Parameters
+        ----------
+
+        info_solver : :class:`snek5000.info.InfoSolverMake`
+
+        classes : iterable of classes
+
+          If a class has the same tag of a default class, it replaces the
+          default one (for example with the tag 'pymech').
+
+        """
+        classesXML = info_solver.classes.PhysFields._set_child("classes")
+
+        avail_classes = [
+            #  pv.ReaderParaview,
+            #  pv.ReaderParaviewStats,
+            pm.ReaderPymech,
+            pm.ReaderPymechStats,
+        ]
+        if classes is not None:
+            avail_classes.extend(classes)
+
+        for cls in avail_classes:
+            classesXML._set_child(
+                cls.tag,
+                attribs={
+                    "module_name": cls.__module__,
+                    "class_name": cls.__name__,
+                },
+            )
+
+    @classmethod
+    def _complete_params_with_default(cls, params, info_solver):
+        params.output._set_child(
+            "phys_fields", attribs={"reader": "pymech", "available_readers": []}
+        )
+
+        dict_classes = info_solver.classes.Output.classes.PhysFields.import_classes()
+        iter_complete_params(params, info_solver, dict_classes.values())
+
+    @cached_property
```

Comment:
It seems to me that it would be better not to use a `cached_property` and to have an initialization step loading a default reader and a method `change_reader` to be able to change on the fly the reader.

Reply:
Done in d99c70243bcf99360fc489b95102c72de89d06cf